src/control/cmd/dmg/pretty/storage_nvme.go

```diff
@@ -164,32 +164,38 @@ func parseNvmeFormatResults(inResults storage.NvmeControllers) storage.NvmeContr
 	parsedResults := make(storage.NvmeControllers, 0, len(inResults))
 	for _, result := range inResults {
 		if result.PciAddr != storage.NilBdevAddress {
-			// ignore skip results
 			parsedResults = append(parsedResults, result)
 		}
 	}
 
 	return parsedResults
 }
 
-func printNvmeFormatResults(devices storage.NvmeControllers, out io.Writer, opts ...PrintConfigOption) error {
-	if len(devices) == 0 {
+func printNvmeFormatResults(ctrlrs storage.NvmeControllers, out io.Writer, opts ...PrintConfigOption) error {
+	if len(ctrlrs) == 0 {
 		fmt.Fprintln(out, "\tNo NVMe devices found")
 		return nil
 	}
 
 	pciTitle := "NVMe PCI"
 	resultTitle := "Format Result"
+	rolesTitle := "Role(s)"
 
-	formatter := txtfmt.NewTableFormatter(pciTitle, resultTitle)
+	formatter := txtfmt.NewTableFormatter(pciTitle, resultTitle, rolesTitle)
 	formatter.InitWriter(out)
 	var table []txtfmt.TableRow
 
-	sort.Slice(devices, func(i, j int) bool { return devices[i].PciAddr < devices[j].PciAddr })
+	sort.Slice(ctrlrs, func(i, j int) bool { return ctrlrs[i].PciAddr < ctrlrs[j].PciAddr })
 
-	for _, device := range parseNvmeFormatResults(devices) {
-		row := txtfmt.TableRow{pciTitle: device.PciAddr}
-		row[resultTitle] = device.Info
+	for _, ctrlr := range parseNvmeFormatResults(ctrlrs) {
+		row := txtfmt.TableRow{pciTitle: ctrlr.PciAddr}
+		row[resultTitle] = ctrlr.Info
+		roles := "???"
```

Comment:
Instead of ???, the role data could be used for md-on-scm ?

Reply:
@Michael-Hennecke for PMem mode I'm displaying N/A for roles Rather than "Data", is that appropriate or should I change?

Reply:
while the NVMe usage clearly is for "data", the PMem mode doesn't have any roles (and we state for the yml that either all 3 roles or none of them shall be assigned). so for consistency this should probably be empty or "NULL" or "n/a". But using "???" as the text string seems odd, because there's nothing unknown about this state.

Reply:
Thanks @Michael-Hennecke , your responses in line with my approach. "???" should never end up being printed, it is simply a placeholder for either "N/A" in PMem mode or roles in MD-on-SSD mode. I didn't want to start with an empty string or "N/A" as the default as neither are useful and both could be confusing. "???" simply implies that it hasn't been set yet and shouldn't end up being displayed to the user.